server/s3_head_object_action.cc

```diff
@@ -74,6 +96,10 @@ void S3HeadObjectAction::fetch_object_info_failed() {
     // There is no object list index, hence object doesn't exist
     s3_log(S3_LOG_DEBUG, request_id, "Object not found\n");
     set_s3_error("NoSuchKey");
+  } else if (!object_metadata || request->has_query_param_key("versionId")) {
+    s3_log(S3_LOG_DEBUG, request_id,
+           "Version-id provided in the request is either invalid or empty\n");
+    set_s3_error("InvalidArgument");
```

Comment:
Below are the responses from aws when object with version if not found - 

```
#aws s3api head-object --bucket "sjain.bucket" --profile aws --key help --version-id ""
An error occurred (400) when calling the HeadObject operation: Bad Request


#aws s3api head-object --bucket "sjain.bucket" --profile aws --key help --version-id "v5n1PPUjOCMHLyyfq0ayjoZ5RZLfVQKA"

An error occurred (404) when calling the HeadObject operation: Not Found

#aws s3api head-object --bucket "sjain.bucket" --profile aws --key help --version-id "dsf"

An error occurred (400) when calling the HeadObject operation: Bad Request

```

Reply:
@sanalpk Please confirm if our implementation matches these responses.

Reply:
yes, But I dont see it in our error code list. even InvalidArgument  it is converting somewhere and i am getting correct response in cli

Reply:
Also please see the st cases written as per aws behavior

Reply:
@sanalpk The CLI might be converting the responses, please try with "--debug" to see the actual response coming from Cortx S3 server.

Reply:
Resonse body is empty
1) invalid/empty version_id 
2021-12-24 05:35:15,516 - MainThread - botocore.parsers - DEBUG - Response headers: {'Content-Length': '0', 'x-amz-request-id': '103ac24b-3b66-4897-9212-3f46743e8c16', 'Content-Type': 'text/plain'}
2021-12-24 05:35:15,516 - MainThread - botocore.parsers - DEBUG - Response body:

2)invalid key
2021-12-24 05:38:52,061 - MainThread - botocore.parsers - DEBUG - Response headers: {'Content-Type': 'application/xml', 'Content-Length': '236', 'x-amz-request-id': '43d18c50-17bb-40b5-a753-1f5c242a7715', 'Server': 'SeagateS3'}
2021-12-24 05:38:52,061 - MainThread - botocore.parsers - DEBUG - Response body:

Reply:
AWS behavior
===========
1)Invalid version_id
Response headers: {'x-amz-request-id': 'FF3H35T08DS3642C', 'x-amz-id-2': 'o8Ec52g4Wq5opj2IIwC7QvLDNQa50ukOaZV2wNk4QTD+dAswOQNO8ZjXwBHPYjVA9XEqjge30Ds=', 'Content-Type': 'application/xml', 'Date': 'Mon, 27 Dec 2021 09:36:33 GMT', 'Server': 'AmazonS3'}
2021-12-27 09:36:34,979 - MainThread - botocore.parsers - DEBUG - Response body:

2)empty version id
Response headers: {'x-amz-request-id': 'SKDZAQFQF46G1XRR', 'x-amz-id-2': 'Uhsu7GJOL7PdAV57ogYOtj8MKRuuVAggEyVODVZY7xTXrbhAkzojsUYpDxPJzKJeu7MYQZm/7Yc=', 'Content-Type': 'application/xml', 'Date': 'Mon, 27 Dec 2021 09:39:54 GMT', 'Server': 'AmazonS3'}
2021-12-27 09:39:55,169 - MainThread - botocore.parsers - DEBUG - Response body:

2)invalid key
s - DEBUG - Response headers: {'x-amz-request-id': '50P5X3NGN5RVNKC9', 'x-amz-id-2': 'pc3bQ6jwPFnfyD+WC37M5uZ3I4a81ZYZ1wQzZC5bHdxXm0yTLXCx+SEzd+t2dF3X0GfgRTCbn2I=', 'Content-Type': 'application/xml', 'Date': 'Mon, 27 Dec 2021 09:40:53 GMT', 'Server': 'AmazonS3'}
2021-12-27 09:40:54,281 - MainThread - botocore.parsers - DEBUG - Response body:


Conclusion: in above cases, aws console also receiving empty  Response body.